python/pyarrow/dataset.py

```diff
@@ -714,9 +729,12 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
         and `format` is not specified, it defaults to the same format as the
         specified FileSystemDataset. When writing a Table or RecordBatch, this
         keyword is required.
-    partitioning : Partitioning, optional
+    partitioning : Partitioning or list[str], optional
         The partitioning scheme specified with the ``partitioning()``
-        function.
+        function or as a list of field names.
+    partitioning_flavor : str, optional
```

Comment:
Might be good to mention the default behavior?  It's kind of hidden in the code but I think its inheriting the default from `pyarrow.dataset.partitioning`

Reply:
Default behaviour is equal to providing `partitioning(pa.schema([])` (haven't changed this).
I would gladly document it, but I'm unsure about what a partitioning with an empty schema means.

From what I can see it works by creating files without any directory. What I'm not sure about is if the data will ever be split in multiple files or if we will always only save `part-0.parquet` because without a partitioning column no chunks will exist.

Reply:
I think Weston meant which partitioning flavor was used by default (directory or hive). And so that is indeed inherited from `partitioning()` -> if not specified, it's directory partitioning, but you can do `partitioning_flavor="hive"` to get hive partitioning.

Reply:
added docs